### PR TITLE
Feature: postgres v11 database(s)

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Migrate send letter DB to version 11](https://tools.hmcts.net/jira/browse/BPS-1057)

### Change description ###

This PR:

- Creates new DB
- Stores it's secrets in vault
- Replaces staging db to v11. It is only used for testing purposes in pipeline. Can safely nuke it and rebuild from scratch

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -96,6 +111,36 @@ resource "azurerm_key_vault_secret" "POSTGRES_DATABASE" {
   value        = "${module.db.postgresql_database}"
 }
 
+resource "azurerm_key_vault_secret" "new_postgres_user" {
+  key_vault_id = "${module.send-letter-key-vault.key_vault_id}"
+  name         = "${var.component}-new-db-user"
```
Comment (nitinprabhu) [inline infrastructure/main.tf:116]: can we plz name as `${var.component}-db-user-v11` so that it's clear what the secret corresponds to ?

Comment (doncem) [reply infrastructure/main.tf:116]: sounds good to me. give me a tic

File: infrastructure/main.tf [hunk 2]
```diff
@@ -96,6 +111,36 @@ resource "azurerm_key_vault_secret" "POSTGRES_DATABASE" {
   value        = "${module.db.postgresql_database}"
 }
 
+resource "azurerm_key_vault_secret" "postgres_user_v11" {
+  key_vault_id = "${module.send-letter-key-vault.key_vault_id}"
+  name         = "${var.component}-db-user-v11"
+  value        = "${module.db.user_name}"
```
Comment (nitinprabhu) [inline infrastructure/main.tf:117]: db-v11 should be used ?

File: infrastructure/main.tf [hunk 3]
```diff
@@ -96,6 +111,36 @@ resource "azurerm_key_vault_secret" "POSTGRES_DATABASE" {
...
+}
+
+resource "azurerm_key_vault_secret" "postgres_pass_v11" {
+  key_vault_id = "${module.send-letter-key-vault.key_vault_id}"
+  name         = "${var.component}-db-pass-v11"
+  value        = "${module.db.postgresql_password}"
```
Comment (nitinprabhu) [inline infrastructure/main.tf:123]: same for other secrets

File: infrastructure/main.tf (hunk 2 above)
Comment (doncem) [reply infrastructure/main.tf:117]: this pr is creating resources so the next pr can be using them via flux config ;)

Comment (nitinprabhu) [reply infrastructure/main.tf:117]: I mean should it be not `"${module.db-v11.user_name}` as you are populating new secrets ?

Comment (doncem) [reply infrastructure/main.tf:117]: oooh got you. yes yes yes! replacing names is not enough with this copy/paste. thanks

**Pull request merged**